components/server/src/api/configuration-service-api.ts

```diff
@@ -105,6 +106,44 @@ export class ConfigurationServiceAPI implements ServiceImpl<typeof Configuration
         });
     }
 
+    async updateConfiguration(req: UpdateConfigurationRequest, _: HandlerContext) {
+        if (!req.configurationId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "configuration_id is required");
+        }
+
+        const userId = ctxUserId();
+        const installer = await this.userService.findUserById(userId, userId);
+        if (!installer) {
+            throw new ApplicationError(ErrorCodes.NOT_FOUND, "user not found");
+        }
+
+        const update: PartialConfiguration = {
+            id: req.configurationId,
+        };
+
+        if (typeof req.name === "string") {
+            update.name = req.name;
+        }
+        if (req.prebuildSettings) {
+            update.prebuildSettings = req.prebuildSettings;
```

Comment:
@filiptronicek we need to do it recursively

Reply:
i..e 
```
const updatePrebuildSettings = {}
if (typeof req.prebuildsSettings?.lalala === "string") {
   updatePrebuildSettings.lalala = req.prebuildsSettings?.lalala;
}
if (Objects.keys(updatePrebuildSettings).length) {
   update.workspaceSettings = updatePrebuildSettings;
}
```

Maybe we need to create a helper function which can do it recursively and used by different clients.